src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreBuildHandler.cs

```diff
@@ -117,6 +120,17 @@ public void Dispose()
 
         public void UpdateSolution_QueryDelayBuildAction(uint dwAction, out IVsTask pDelayTask)
         {
+            if (!_isMEFInitialized)
+            {
+                ThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
Except these MEF initialization, there is nothing else happens as part of this package load. Besides these MEF imports also doesn't have any other opportunity to improve performance since they are already very minimalistic. Although there are few VS platform MEF imports as well which I'm not sure whether have any further scope of improvement or not but I'll very doubtful of that as well.